core/src/main/java/org/apache/shiro/realm/text/TextConfigurationRealm.java

```diff
@@ -211,9 +215,15 @@ protected void processUserDefinitions(Map<String, String> userDefs) {
 
     protected static Set<String> toLines(String s) {
         LinkedHashSet<String> set = new LinkedHashSet<String>();
-        Scanner scanner = new Scanner(s);
-        while (scanner.hasNextLine()) {
-            set.add(scanner.nextLine());
+        try (Scanner scanner = new Scanner(s)) {
+            while (scanner.hasNextLine()) {
+                set.add(scanner.nextLine());
+            }
+        } catch (Exception e) {
+            if (log.isWarnEnabled()) {
+                String msg = "Unable to fetch next line, Scanner stream corrupted.";
+                log.warn(msg, e);
+            }
```

Comment:
Do not catch a generic exception. In this case, do not catch anything at all. We would want the Exception to promote here, otherwise the behaviour would change.

Reply:
Agree, we want the original exception thrown here, failure to parse the config (especially config related to security) is a valid reason to throw an exception and resulting in a failure of the application to start.

A couple of other nits, as you learn more about Java:

you don't need to check `log.isWarnEnabled()` unless you are concatenating strings, or doing another expensive operation in that block, the log back `warn` method will do this check for you. 

But like we mentioned, we don't want to swallow this exception

@pitjazz let's close this PR for now, if there is a listing issue you want to fix, we can continue the thread here and provide other suggestions if needed